Decred Wallet/Features/Navigation Menu/NavigationMenuBaseController.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  NavigationMenuBaseController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+class NavigationMenuBaseController: UITabBarController {
+    
+    var isNewWallet: Bool = false
+    var floatingButtons: NavMenuFloatingButtons!
+    var customTabBar: TabMenu!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.loadTabBar()
+        self.loadFloatingButtons() // TODO: Organize this to only show on the two pages where it is needed
+        if self.isNewWallet {
+            self.showOkAlert(message: LocalizedStrings.newWalletMsg)
+        }
+    }
+    
+    func loadTabBar() {
+        let tabItems: [MenuItem] = [.overview, .transactions, .accounts, .more]
+        
+        self.setupCustomTabMenu(tabItems){ (controllers) in
+            self.viewControllers = controllers
+        }
+        self.selectedIndex = 0
+    }
+    
+    public func loadFloatingButtons() {
+        self.floatingButtons = NavMenuFloatingButtons()
+        self.floatingButtons.translatesAutoresizingMaskIntoConstraints = false
+        self.floatingButtons.clipsToBounds = true
+        
+        self.view.addSubview(self.floatingButtons)
+        let constraints = [
+            self.floatingButtons.heightAnchor.constraint(equalToConstant: 48),
+            self.floatingButtons.widthAnchor.constraint(equalToConstant: 240),
+            self.floatingButtons.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
+            self.floatingButtons.bottomAnchor.constraint(equalTo: customTabBar.topAnchor, constant: -12),
+        ]
+        NSLayoutConstraint.activate(constraints)
+    }
+    
+    func setupCustomTabMenu(_ withItems: [MenuItem], completion: @escaping ([UIViewController]) -> Void) {
+        let frame = tabBar.frame
+        var controllers = [UIViewController]()
+        
+        self.customTabBar = TabMenu(items: withItems, frame: frame)
+        self.customTabBar.clipsToBounds = true
+        
+        view.addSubview(customTabBar)
+        self.customTabBar.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            customTabBar.leadingAnchor.constraint(equalTo: tabBar.leadingAnchor),
+            customTabBar.trailingAnchor.constraint(equalTo: tabBar.trailingAnchor),
+            customTabBar.widthAnchor.constraint(equalToConstant: tabBar.frame.width),
+            customTabBar.heightAnchor.constraint(equalToConstant: 67),
+            customTabBar.bottomAnchor.constraint(equalTo: tabBar.bottomAnchor)
+        ])
+        
+        for i in 0 ..< withItems.count {
+            controllers.append(withItems[i].viewController)
+        }
+        
+        self.customTabBar.itemTapped.subscribe(with: self){ (index) in
+            DispatchQueue.main.async {
+                self.selectedIndex = index
```

Comment:
Don't see a use for this variable yet, also don't need an observable for this. Should instead add a convenience method to `customTabBar` to read the selected index whenever necessary.

Reply:
We want to react to item taps the moment they happen. the tab bar items fire a Signal event (in this case, the `itemTapped` variable) when tapped and send their `tag` properties with the event. this lets us attach and forget so we can react in realtime. another approach would have been a `tabNavigationMenuItemTapped` delegate but that would require more lines of code on both `NavigationMenuBaseController` and `TabNavigationMenu`. this keeps it short and in just one place.